pandas/core/groupby/generic.py

```diff
@@ -1416,6 +1424,17 @@ def aggregate(self, func=None, *args, engine=None, engine_kwargs=None, **kwargs)
             result.columns = columns  # type: ignore[assignment]
 
         if result is None:
+            # Remove the kwargs we inserted
```

Comment:
So we need to go through `result = op.agg()` first instead of just using `self._aggregate_with_numba` in the `if if maybe_use_numba(engine)` branch above?